draft-hzpa-dprive-xfr-over-tls.md

```diff
@@ -148,69 +149,408 @@ DoT: DNS-over-TLS as specified in [@!RFC7858]
+EDNS0 Keepalive [RFC7828].
+
+## AXoT Sessions
+
+The figure below provides an outline of the AXoT mechanism including NOTIFYs.
+
+[Figure 3: AXoT mechanism](https://docs.google.com/drawings/d/1EGNU7nlumBgxu9r-O1aoETmlzuOB4H6VcxrfVl2ffwU/edit?usp=sharing)
+
+All implementations that support XoT MUST fully implement [@RFC5953] behavior
+on TLS connections.
+
+Sections 4.1, 4.1.1 and 4.1.2 of [@RFC5936] describe guidance for AXFR clients
+and servers with regard to re-use of sessions for multiple AXFRs, AXFRs of
+different zones and using TCP session for other queries including SOA. 
+
+The connection for AXFR-over-TLS SHOULD be established using port 853, as
+specified in [@!RFC7858], unless there is mutual agreement between the secondary
+and primary to use a port other than port 853 for XFR-over-TLS.
+
+QUESTION: Should we require that the SOA is always done on a TLS connection?
+For the case when no transfer is required this could be unnecessary overhead.
+
+## IXoT Requests
+
+The figure below provides an outline of the XoT mechanism including NOTIFYs.
+
+[Figure 4: IXoT mechanism]
+(https://docs.google.com/drawings/d/1yyDtyqOJogZzlZ6bpf9nx2h_9zeXXG6Q9yla_502jFc/edit?usp=sharing)
+
+The connection for IXFR-over-TLS SHOULD be established using port 853, as
+specified in [@!RFC7858], unless there is mutual agreement between the secondary
+and primary to use a port other than port 853 for XFR-over-TLS.
+
+[@RFC1995] says nothing with respect to optimizing IXFRs over TCP or re-using
+already open TCP connections to perform IXFRs or other queries. We provide
+guidance here that aligns with the guideance in [@RFC5936] for AXFR and with
+that for performant TCP/TLS usage in [@RFC7766] and [@RFC7858].
+
+An IXoT client MAY use an already opened TLS connection to send a IXFR request.
+Using an existing open connection is RECOMMENDED over opening a new connection.
+(Non-IXoT session traffic can also use an open connection.)
+
+An IXoT client MAY use an already open TLS connection to send an SOA query.
+Using an existing open connection is RECOMMENDED over opening a new connection.
+
+An IXoT server MUST be able to handle multiple IXoT requests on a single TLS
+connection, as well as to handle other query/response transactions over it.
+
+An IXoT client MAY keep an existing TLS session open in the expectation it is
+likely to need to perform an IXFR in the near future. The client may use the
+frequency of recent IXFRs to calculate an average update rate and then use
+EDNS0 Keepalive to request an appropriate timeout from the server (if the
+server supports EDNS0 Keepalive).
+
+An IXoT client MAY pipeline IXFR requests for different zones on a single TLS
+connection. AN XIoT server MAY respond to those requests out of order.
+
+### Performance
+
+In workloads where there are frequent IXFRs, is the persistent connection mode
+that TCP-Mode would enable (as well as the retries) a benefit?
```

Comment:
Not if it requires the NOTIFY, SOA query and then (A|I)XFR query interaction I think.
Only with DSO subscription scheme Nagle's algorithm will help performance on high load IXFR streams (as far as I can see)